### DISCUSSION TITLE
Micropython is not flashing ESP32-S3-WROOM-2-N32R8V

### DISCUSSION THREAD

**Discussion by gaurav0096** [micropython/micropython#12103] [ESP32]
i am trying to flash micropython firmware on this dev board ESP32-S3-WROOM-2-N32R8V, 32 MB Octal flash & 8 MB Octal PSRAM.
Specification of board
esptool.py v4.7-dev
Serial port COM42
Connecting....
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (QFN56) (revision v0.1)
Features: WiFi, BLE, Embedded PSRAM 8MB (AP_1v8)
Crystal is 40MHz
MAC: 7c:df:a1:e4:6e:a4
Uploading stub...
Running stub...
Stub running...
Manufacturer: c2
Device: 8039
Detected flash size: 32MB
Flash type set in eFuse: octal (8 data lines)
Hard resetting via RTS pin..

I am getting below error

ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x18 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
Octal Flash Mode Enabled
For OPI Flash, Use Default Flash Boot Mode
mode:SLOW_RD, clock div:1
load:0x3fce3808,len:0xf34
load:0x403c9700,len:0xc44
load:0x403cc700,len:0x2c78
entry 0x403c98f4
E (31) qio_mode: Failed to set QIE bit, not enabling QIO mode
Traceback (most recent call last):
  File "_boot.py", line 11, in <module>
  File "inisetup.py", line 34, in setup
  File "inisetup.py", line 7, in check_bootsec
OSError: (-24579, 'ESP_ERR_FLASH_NOT_INITIALISED')
MicroPython v1.20.0 on 2023-04-26; ESP32S3 module (spiram octal) with ESP32S3
Type "help()" for more information.

**Comment by sophiedegran** [comment]
Hi,
There is no specific micropython image for the N32R8 Espressif S3 board.
You have to make it by yourself with some param values you'll find on previous topics.
This  board work very well and is really a speeder board ;)

**Reply by gaurav0096** [reply]
Thanks for the quick reply @sophiedegran , i am very new to micropython, can you please help me to generate micropython image N32R8 using windows 10.

**Comment by scruss** [comment]
Maybe try the most recent nightly from [Generic ESP32-S3 (SPIRAM Octal)](https://micropython.org/download/GENERIC_S3_SPIRAM_OCT/): it's most likely the bin file you want, rather than the app-bin (though I could be wrong).

The v1.20.0 build wasn't compiled with ESP-IDF v5, while the nightlies are

**Reply by gaurav0096** [reply]
Thanks for reply, it didn't work i have tried nightly builds

**Comment by sophiedegran** [comment]
Under windows 10 you have to install wsl and Ubuntu 20.04
After that see previous post (search for N32R8)
I use git for espressif 4.4
Take your time it's very simple .. and when you've done one it's very simple to compil

**Reply by robert-hh** [reply]
The actual version of the MicroPython code for ESP32 builds with ESP-IDF v5.0.2. It does not support previous versions of the IDF any more.

**Reply by sophiedegran** [reply]
Thanks robert but on stable version 1.20 i use esp-idf v4.4

**Reply by robert-hh** [reply]
That's right. But if you build from the tip of the actual master branch, you have to use v5.0.2 at least. Obviously, if you check out v1.20.0, you have to use v4.4.x or v4.2.x.

**Reply by sophiedegran** [reply]
Yes of course! Many thanks for your git and all you've done it's very instructive!

**Reply by gaurav0096** [reply]
thanks @robert-hh
i have searched but i am not found exact post which will help to build micropython image.

**Reply by sophiedegran** [reply]
See discussion 10359 
For windows 10:
- install wsl
- install ubuntu
- install dependencies for esp
- git clone micropython
- git clone espressif v5.0.2 (for night build micropython) or v4.4 for stable version 1.20
- try your first make on micropython with mpy-cross (that's for mpy files)
cd micropython -> make -C mpy-cross
If ok continue as explain in discussion 10359
I'm sure you will do it !

**Reply by gaurav0096** [reply]
thanks @sophiedegran i have followed the instruction 10359
now i am getting below error, after selecting from menuconfig 

Error:
No changes to save (for '/home/gaurav/esp/mpy/micropython/ports/esp32/build-GENERIC_S3_SPIRAM/sdkconfig')
Loading defaults file /home/gaurav/esp/mpy/micropython/ports/esp32/build-GENERIC_S3_SPIRAM/sdkconfig.combined...
/tmp/confgen_tmpw5af11vw:108 line was updated to CONFIG_ESPTOOLPY_FLASHSIZE_4MB=n
/tmp/confgen_tmpw5af11vw:110 line was updated to CONFIG_ESPTOOLPY_FLASHSIZE_16MB=n
Traceback (most recent call last):
  File "makeimg.py", line 59, in <module>
    partition_table = load_partition_table(arg_partitions_bin)
  File "makeimg.py", line 37, in load_partition_table
    with open(filename, "rb") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'build-GENERIC_S3_SPIRAM/partition_table/partition-table.bin'
make: * [Makefile:57: all] Error 1

**Reply by sophiedegran** [reply]
Ok that's good! Don't use menuconfig just edit the file sdkconfig.board and do changes by yourself ...
check also sdkconfig.spiram_sx
verify also what mpconfigboard.cmake will do ..
You are with esp-idf 5.0.2 ? seem he can't find 'partitions-32MiB.csv'

**Reply by gaurav0096** [reply]
Thanks @sophiedegran its working

**Reply by sophiedegran** [reply]
You are the boss! You make it on esp-idf 5.0.2 ?

**Reply by gaurav0096** [reply]
i have used esp-idf v5.0.3.

**Comment by bidrohini** [comment]
I can see another thread regarding this OSError: (-24579, 'ESP_ERR_FLASH_NOT_INITIALISED') message:

 https://github.com/orgs/micropython/discussions/10156

**Comment by andypiper** [comment]
I have a ESP32-S3-MINI-N32R8V and I had a similar issue with the `-24579, 'ESP_ERR_FLASH_NOT_INITIALISED'` error when using one of the builds available from micropython.org.

I was able to compile a custom build (using IDF 5.0.2 on macOS) - primarily I needed to modify the build to have the correct amount of flash (in this case it was `CONFIG_ESPTOOLPY_FLASHSIZE_4MB=y` for the mini).

**Comment by sophiedegran** [comment]
Andy
I don't know these board Esp32s3 Mini N32r8. There are a lot 'clone' from espressif and that is a shame..
Normally, N32 mean 32mo so flashsize = 32MB and give the partition custom file name = partitions-32MiB.csv
and of course declare this:
CONFIG_ESPTOOLPY_OCT_FLASH=y
CONFIG_ESPTOOLPY_FLASHMODE="dout"

First thing is to determine if your board is spiram or octal or boths!
Hope that help

**Reply by andypiper** [reply]
Apologies for the confusion, and thank you for the information about the way this numbering works!

My post above was incorrect - I have a ESP32-S3-MINI-N4R2 board, which is why `CONFIG_ESPTOOLPY_FLASHSIZE_4MB=y` was the correct configuration.

**Comment by Amundw99** [comment]
How do you actually edit the sdkconfig file? Notepad editor?

**Reply by sophiedegran** [reply]
Which board do you have and the micropython version you want to install?

**Reply by Amundw99** [reply]
ESP32-S3-DEVKITC-1-N32R8V

**Comment by halherta** [comment]
I have a [tutorial](https://hussamtalkstech.com/p/building-a-micropython-image-from-source-for-the-esp32-s3-with-16mb-of-flash/) on how to create an image for the ESP32-S3 Module with 16MB of flash. To do the same for the module with 32MB of flash, repeat the steps for 32MB. Pretty sure a 'partitions-32MiB.csv' exists in the MicroPython source tree where all the partition files are saved.
